src/hooks/useSyncChainQuery.ts

```diff
@@ -31,8 +41,23 @@ export default function useSyncChainQuery() {
   const [searchParams, setSearchParams] = useSearchParams()
 
   useEffect(() => {
-    if (isActive && urlChainId && chainId !== urlChainId) {
+    // Change a user's chain on pageload if the connected chainId does not match the query param chain
+    if (isActive && urlChainId && chainIdRef.current === chainId && chainId !== urlChainId) {
```

Comment:
Isn't `isActive` the same as `account`? If so, could you just return early in this useEffect to simplify a bit: `if (!account) return`?

Reply:
No `isActive` can be true if `account` is undefined

Reply:
That was actually the issue with the Celo bug you pointed out earlier, I had been using account as a check but it doesn't work when no wallet is connected so I switched to isActive

Reply:
Commit for that fix for reference https://github.com/Uniswap/interface/pull/6926/files/b26d1d52477a08e6ba662a0be543274c55fc91d9..db314e2da203ea8a040e2a5852d0f54f6e269651